cloudsync_onedrive.py

```diff
@@ -447,13 +447,14 @@ def _fetch_shared_drives(self):
     def _fetch_sites(self):
         # sharepoint sites - a user can have access to multiple sites, with multiple drives in each
         sites = self._direct_api("get", "/sites?search=*").get("value", [])
-        sites.sort(key=lambda s: s["displayName"].lower())
+        sites.sort(key=lambda s: (s.get("displayName") or s.get("name", "")).lower())
         for site in sites:
             try:
                 # TODO: use configurable regex for filtering?
                 url_path = urllib.parse.unquote_plus(urllib.parse.urlparse(site["webUrl"]).path).lower()
                 if not url_path.startswith("/portals/"):
-                    self.__site_by_id[site["id"]] = Site(name=site["displayName"], id=site["id"])
+                    name = site.get("displayName") or site.get("name", "")
```

Comment:
Could throw this in a property if you wanted, LGTM though

Reply:
this could use some refactoring, looking to review/fix all of our JSON-ingestion code in a separate PR.

at this point `site` is just a dict though, so can't easily add a property to it